eng/Directory.Build.Data.props

```diff
@@ -26,6 +26,12 @@
       CA2237; <!-- [Serializeable] on exceptions -->
       CA2225; <!-- Provide alternative to implicit operators -->
       CA1714; <!-- Flags should have plural names -->
+      AZC0106; <!-- Temporary disabled: Non-public asynchronous method needs 'async' parameter. -->
```

Comment:
Did you intend to disable these for all libraries?

Reply:
Yes. These diagnostics are new, they aren't published yet: https://github.com/Azure/azure-sdk-tools/pull/375. I'm going to re-enable them in global props during phase 2.